connector/json-test-connector/src/common.rs

```diff
@@ -0,0 +1,93 @@
+use fluvio::{Offset, FluvioConfig, metadata::topic::TopicSpec};
+use fluvio_connector_package::config::ConnectorConfig;
+use fluvio_smartengine::transformation::TransformationConfig;
+use futures::stream::LocalBoxStream;
+use anyhow::Result;
+use async_trait::async_trait;
+
+#[async_trait]
+pub(crate) trait Source<'a, I> {
+    async fn connect(self, offset: Option<Offset>) -> Result<LocalBoxStream<'a, I>>;
+}
+
+pub(crate) async fn create_producer(
+    config: &ConnectorConfig,
+) -> anyhow::Result<fluvio::TopicProducer> {
+    let mut cluster_config = FluvioConfig::load()?;
+    cluster_config.client_id = Some(format!("fluvio_connector_{}", &config.name));
+
+    let fluvio = fluvio::Fluvio::connect_with_config(&cluster_config).await?;
+    ensure_topic_exists(config).await?;
+    let mut config_builder = fluvio::TopicProducerConfigBuilder::default();
+
+    if let Some(producer_params) = &config.producer {
+        // Linger
+        if let Some(linger) = producer_params.linger {
+            config_builder = config_builder.linger(linger)
+        };
+
+        // Compression
+        if let Some(compression) = producer_params.compression {
+            config_builder = config_builder.compression(compression)
+        };
+
+        // Batch size
+        if let Some(batch_size) = producer_params.batch_size {
+            config_builder = config_builder.batch_size(batch_size.as_u64() as usize)
+        };
+    };
+
+    let producer_config = config_builder.build()?;
+    let producer = fluvio
+        .topic_producer_with_config(&config.topic, producer_config)
+        .await?;
+
+    if let Some(chain) = create_smart_module_chain(config).await? {
+        Ok(producer.with_chain(chain)?)
+    } else {
+        Ok(producer)
+    }
+}
+pub(crate) async fn ensure_topic_exists(config: &ConnectorConfig) -> anyhow::Result<()> {
+    let admin = fluvio::FluvioAdmin::connect().await?;
+    let topics = admin.list::<TopicSpec, String>(vec![]).await?;
```

Comment:
can use filter argument in the `list` API:  rather than retrieve all and filter

Reply:
done